### ISSUE TITLE
Assess impact of bpo-22708 in our library and test suite. 

### ISSUE THREAD

**Issue by jalopezsilva** [urllib3/urllib3#2750]
See https://github.com/python/cpython/pull/8305 for context. 

The HTTP CONNECT message will be upgraded to HTTP/1.1 from HTTP/1.1. I expect a small set of changes in the library itself.

**Comment by graingert** [comment]
This got merged 2 weeks ago

**Comment by pquentin** [comment]
Yes, but it barely missed 3.12.0 alpha7 that we're using now through GitHub Actions. The commit will be in 3.12.0 beta1 next month, at which point we will start seeing failures, if any.
